__tests__/hooks/headless/useChatMessages.test.tsx

```diff
@@ -509,4 +513,165 @@ describe("useChatMessages", () => {
+      payload: [{ contentString: "Test received status" }],
+      partOf: [createReference(mockThread)],
+    };
+    await medplum.createResource(newMessage);
+
+    // Create and emit the subscription bundle
+    const bundle = await createCommunicationSubBundle(newMessage);
+    act(() => {
+      subManager.emitEventForCriteria(`Communication?part-of=Communication/test-thread`, {
+        type: "message",
+        payload: bundle,
+      });
+    });
+
+    // Verify received timestamp is set
+    await waitFor(() => {
+      const lastMessage = result.current.messages[result.current.messages.length - 1];
+      expect(lastMessage.text).toBe("Test received status");
+      expect(lastMessage.received).toBeDefined();
+      expect(lastMessage.read).toBe(false);
+    });
+  });
+
+  test("Read status is set when markMessageAsRead is called", async () => {
+    const { medplum } = await setup();
+    const { result } = renderHook(() => useChatMessages({ threadId: "test-thread" }), {
+      wrapper: ({ children }) => <MedplumProvider medplum={medplum}>{children}</MedplumProvider>,
+    });
+
+    // Wait for initial load
+    await waitFor(() => {
+      expect(result.current.loading).toBe(false);
+    });
+
+    // Get an unread message
+    const unreadMessage = result.current.messages.find((m) => !m.read);
+    expect(unreadMessage).toBeDefined();
+
+    // Mark message as read
+    await act(async () => {
+      await result.current.markMessageAsRead(unreadMessage!.id);
+    });
+
+    // Verify message is marked as read
+    await waitFor(() => {
+      const message = result.current.messages.find((m) => m.id === unreadMessage!.id);
+      expect(message?.read).toBe(true);
+    });
+  });
+
+  test("markMessageAsRead does nothing if message is already read", async () => {
+    const { medplum } = await setup();
+
+    // Create a new message from practitioner with read status (completed)
+    const newMessage: Communication = {
+      resourceType: "Communication",
+      id: "msg-3",
+      status: "completed",
+      sent: new Date().toISOString(),
+      sender: createReference(mockPractitioner),
+      payload: [{ contentString: "Test received status" }],
+      partOf: [createReference(mockThread)],
+    };
+    await medplum.createResource(newMessage);
+
+    // Render the hook
+    const { result } = renderHook(() => useChatMessages({ threadId: "test-thread" }), {
+      wrapper: ({ children }) => <MedplumProvider medplum={medplum}>{children}</MedplumProvider>,
+    });
+
+    // Wait for initial load
+    await waitFor(() => {
+      expect(result.current.loading).toBe(false);
+    });
+
+    // Mark message as read
+    await act(async () => {
+      await result.current.markMessageAsRead(newMessage.id!);
+    });
+
+    // Verify message is still marked as read
+    await waitFor(() => {
+      const message = result.current.messages.find((m) => m.id === newMessage.id);
+      expect(message?.read).toBe(true);
+    });
+  });
```

Comment:
Maybe also assert the `read` value before the `markMessageAsRead`? Something like:

```suggestion
  test("markMessageAsRead does nothing if message is already read", async () => {
    const { medplum } = await setup();

    // Create a new message from practitioner with read status (completed)
    const newMessage: Communication = {
      resourceType: "Communication",
      id: "msg-3",
      status: "completed",
      sent: new Date().toISOString(),
      sender: createReference(mockPractitioner),
      payload: [{ contentString: "Test received status" }],
      partOf: [createReference(mockThread)],
    };
    await medplum.createResource(newMessage);

    // Render the hook
    const { result } = renderHook(() => useChatMessages({ threadId: "test-thread" }), {
      wrapper: ({ children }) => <MedplumProvider medplum={medplum}>{children}</MedplumProvider>,
    });

    // Wait for initial load
    await waitFor(() => {
      expect(result.current.loading).toBe(false);
      const message = result.current.messages.find((m) => m.id === newMessage.id);
      expect(message?.read).toBe(true);
    });

    // Mark message as read
    await act(async () => {
      await result.current.markMessageAsRead(newMessage.id!);
    });

    // Verify message is still marked as read
    await waitFor(() => {
      const message = result.current.messages.find((m) => m.id === newMessage.id);
      expect(message?.read).toBe(true);
    });
  });
```